src/Npgsql/NpgsqlConnector.cs

```diff
@@ -481,14 +485,7 @@ string GetUsername()
             if (username?.Length > 0)
```

Comment:
I'm not sure anymore whether we should look at Environment.UserName before or after the Kerberos/Windows-specific check. It's probably fine the way it is.

Reply:
Stephen Toub worked in this area and marked everything as nullable if it can be `null`. Looks like `UserName` cannot be `null` and, therefore, should be checked last.

Reply:
Hmm... I'm still not totally sure (but this is not related to this PR). We're basically saying that the Kerberos username takes precedence over the local system one, and I'm not 100% sure that's always the correct behavior. But let's leave it for now.